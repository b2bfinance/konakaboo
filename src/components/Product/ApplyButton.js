import styled from 'styled-components';
import Fab from '@material-ui/core/Fab';

export default styled(Fab)`
  background-color: ${props => props.theme.applyButtonBackground} !important;
  color: ${props => props.theme.applyButtonColor};
  width: 100% !important;
  box-shadow: none !important;

  &:hover {
    color: ${props => props.theme.applyButtonColor};
  }
`;
