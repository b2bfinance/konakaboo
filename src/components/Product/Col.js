import React from 'react';
import styled from 'styled-components';
import Col from '../Col';

export const Wrapper = styled(Col)`
  align-items: center;
  background-color: ${props =>
    props.background ? props.theme.productColBackground : ''};
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 1rem 1.25rem;
  text-align: center;

  img {
    flex-shrink: 0;
    max-width: 100%;
    max-height: 120px;
  }

  span {
    font-size: 1.25rem;
    font-weight: ${props => props.theme.mainBoldFontWeight};
    margin-bottom: 0.5rem;
    line-height: 1.1;
  }

  small {
    font-size: 95%;
    font-weight: ${props => props.theme.mainNormalFontWeight};
  }
`;

export const MultiCol = ({ columns, theme }) =>
  columns.map((column, i) => (
    <Wrapper
      key={i}
      phone="100"
      desktop={60 / columns.length}
      background={(i + 1) % 2}
      theme={theme}
    >
      <small>{column.label}</small>
      <span>{column.value}</span>
    </Wrapper>
  ));

export default Wrapper;
