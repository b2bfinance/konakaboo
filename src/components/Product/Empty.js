import React from 'react';
import styled from 'styled-components';

export const Wrapper = styled.div`
  background-color: ${props => props.theme.productEmptyBackground};
  margin: 2rem auto;
  padding: 2rem;
`;

export default () => (
  <Wrapper>
    <p>There are no product matching your requirements.</p>
  </Wrapper>
);
