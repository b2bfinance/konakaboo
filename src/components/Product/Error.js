import React from 'react';
import styled from 'styled-components';

export const Wrapper = styled.div`
  background-color: #ff9800;
  color: #fff;
  padding: 1rem;
`;

export default ({ message }) => <Wrapper>{message}</Wrapper>;
