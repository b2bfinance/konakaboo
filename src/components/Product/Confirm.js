import React from 'react';
import styled from 'styled-components';
import Button from '../Button';
import Row from '../Row';
import Col from '../Col';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';

export const ConfirmButtonCol = styled(Col)`
  margin: 1.5rem auto 0;
`;

export const ConfirmDialog = styled(Dialog)`
  text-align: center;
`;

const ProductConfirm = ({
  open,
  handleRequestClose,
  title,
  description,
  forwardUrl
}) => (
  <ConfirmDialog open={open} onClose={handleRequestClose}>
    <DialogTitle>{title}</DialogTitle>
    <DialogContent>
      <DialogContentText>{description}</DialogContentText>
      <Row>
        <ConfirmButtonCol phone={80} tablet={60} desktop={40}>
          <Button primary href={forwardUrl}>
            Apply
          </Button>
        </ConfirmButtonCol>
      </Row>
    </DialogContent>
  </ConfirmDialog>
);

export default ProductConfirm;
