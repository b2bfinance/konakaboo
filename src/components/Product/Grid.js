import React from 'react';
import styled from 'styled-components';
import Typography from '@material-ui/core/Typography';
import Col from '../Col';
import { queries } from '../../utils/media';

export const Wrapper = styled(Col)`
  border-top: 2px solid ${props => props.theme.productBorder};
  padding: 16px 8px;
  text-align: center;

  ${queries.tablet`
    margin-top: 16px;
    margin-bottom: 16px;
    padding: 8px;
    border-top: 0;
    border-left: 2px solid ${props => props.theme.productBorder};
  `};

  &:first-child {
    border-top: 0;
    border-left: 0;
  }
`;

export const ProductMultiGrid = ({ columns, theme }) =>
  columns.map((column, i) => (
    <Wrapper
      key={i}
      background={(i + 1) % 2}
      theme={theme}
      item
      phone={100}
      tablet={80 / columns.length}
    >
      <Typography variant="subtitle2" component="span">
        {column.label}
      </Typography>
      <Typography variant="h5" component="span" color="textSecondary">
        {column.value}
      </Typography>
      {column.subtext && (
        <Typography variant="caption">{column.subtext}</Typography>
      )}
    </Wrapper>
  ));

export default Wrapper;
