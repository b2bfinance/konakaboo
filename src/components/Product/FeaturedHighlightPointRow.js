import React from 'react';
import styled from 'styled-components';
import GradeOutline from '@material-ui/icons/GradeOutlined';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';

export const Wrapper = styled(Grid)`
  background-color: ${props => props.theme.productBorder};
  border-top: 2px solid ${props => props.theme.productBorder};
  padding: 8px 16px;
`;

export const HighlightIcon = styled.span`
  color: ${props => props.theme.productFeaturedIconColor};
  font-size: 1.25rem;
`;

export const Feature = styled.span`
  margin-left: 4px;
`;

export default ({ feature }) => (
  <Wrapper container alignItems="center">
    <Grid item>
      <HighlightIcon>
        <GradeOutline fontSize="inherit" />
      </HighlightIcon>
    </Grid>
    <Grid item>
      <Typography variant="subtitle1">
        <Feature>{feature}</Feature>
      </Typography>
    </Grid>
  </Wrapper>
);
