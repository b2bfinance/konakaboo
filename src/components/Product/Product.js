import React, { useState } from 'react';
import styled, { css } from 'styled-components';
import Grid from '@material-ui/core/Grid';
import Confirm from './Confirm';
import MoreInfo from './MoreInfo';
import HeadingRow from './HeadingRow';
import ProductGrid, { ProductMultiGrid } from './Grid';
import MoreInfoButton from './MoreInfoButton';
import ApplyButton from './ApplyButton';
import FeaturedHighlightPointRow from './FeaturedHighlightPointRow';
import Typography from '@material-ui/core/Typography';

export const Wrapper = styled.div`
  margin-bottom: 56px;
`;

export const ProductBody = styled(Grid)`
  border: 2px solid ${props => props.theme.productBorder};
  background-color: ${props => props.theme.productBackground};

  ${props =>
    props.standOut &&
    css`
      border-color: ${props.theme.productHighlightBorder};
    `};
`;

export const ProductDescription = styled(Typography)`
  border-top: 2px solid ${props => props.theme.productBorder};
  padding: 24px 16px;
  width: 100%;
`;

export const Product = ({
  cta,
  highlighted,
  labels,
  title,
  links,
  brand,
  columns,
  description,
  feature_point,
  detailed,
  disclaimer,
  meta
}) => {
  const [withInfo, setWithInfo] = useState(false);
  const [withConfirmation, setWithConfirmation] = useState(false);

  return (
    <Wrapper>
      <HeadingRow
        logo={links.logo}
        brand={brand}
        highlighted={highlighted}
        title={title}
        labels={labels}
      />
      <ProductBody container>
        <ProductMultiGrid columns={columns} />
        <ProductGrid phone={100} tablet={20}>
          <Grid container alignContent="center">
            <Grid item xs={12}>
              {detailed && (
                <MoreInfoButton
                  onClick={() => setWithInfo(!withInfo)}
                  component="button"
                >
                  more details
                </MoreInfoButton>
              )}
            </Grid>
            <Grid item xs={12}>
              <ApplyButton
                variant="extended"
                color="primary"
                size="medium"
                href={links.apply}
                target="_blank"
                rel="noopener"
                onClick={
                  meta.confirm
                    ? e => e.preventDefault() || setWithConfirmation(true)
                    : undefined
                }
              >
                {cta}
              </ApplyButton>
            </Grid>
          </Grid>
        </ProductGrid>
        {description && (
          <ProductDescription variant="body2">{description}</ProductDescription>
        )}
        {feature_point && <FeaturedHighlightPointRow feature={feature_point} />}
      </ProductBody>
      {detailed && (
        <MoreInfo
          open={withInfo}
          onClose={() => setWithInfo(false)}
          title={title}
          links={links}
          brand={brand}
          description={description}
          detailed={detailed}
          disclaimer={disclaimer}
        />
      )}
      {meta.confirm && (
        <Confirm
          open={withConfirmation}
          handleRequestClose={() => setWithConfirmation(false)}
          title={meta.confirm.heading}
          description={meta.confirm.description}
          forwardUrl={links.apply}
        />
      )}
    </Wrapper>
  );
};

export default Product;
