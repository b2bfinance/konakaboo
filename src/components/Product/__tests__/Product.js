import React from 'react';
import { shallow, render } from 'enzyme';
import Button from '../../Button';
import Product, {
  Container,
  HeadingRow,
  StyledProductLabels,
  ProductCol,
  MoreInfoRow,
  ApplyButton,
  InfoList,
  HighlightPoint,
  TechnicalPoint,
  ProductLabels,
  ProductColumns,
  ProductHighlightPoints,
  ProductTechnicalPoints
} from '../Product';

test('Container renders correctly', () => {
  expect(render(<Container theme={stubData.theme} />)).toMatchSnapshot();
});

test('HeadingRow renders correctly', () => {
  expect(render(<HeadingRow theme={stubData.theme} />)).toMatchSnapshot();
});

test('StyledProductLabels renders correctly', () => {
  expect(render(<StyledProductLabels />)).toMatchSnapshot();
});

test('ProductCol renders correctly', () => {
  expect(render(<ProductCol theme={stubData.theme} />)).toMatchSnapshot();
});

test('MoreInfoRow renders correctly', () => {
  expect(render(<MoreInfoRow theme={stubData.theme} />)).toMatchSnapshot();
});

test('ApplyButton renders correctly', () => {
  expect(render(<ApplyButton />)).toMatchSnapshot();
});

test('InfoList renders correctly', () => {
  expect(render(<InfoList />)).toMatchSnapshot();
});

test('HighlightPoint renders correctly', () => {
  expect(render(<HighlightPoint theme={stubData.theme} />)).toMatchSnapshot();
});

test('TechnicalPoint renders correctly', () => {
  expect(render(<TechnicalPoint theme={stubData.theme} />)).toMatchSnapshot();
});

test('ProductLabels renders correctly', () => {
  expect(render(<ProductLabels labels={['test']} />)).toMatchSnapshot();
});

test('ProductColumns renders correctly', () => {
  expect(
    render(<ProductColumns columns={[{ label: 'test', value: 'test' }]} />)
  ).toMatchSnapshot();
});

test('ProductHighlightPoints renders correctly', () => {
  expect(
    render(<ProductHighlightPoints points={['test']} />)
  ).toMatchSnapshot();
});

test('ProductTechnicalPoints renders correctly', () => {
  expect(
    render(<ProductTechnicalPoints points={['test']} />)
  ).toMatchSnapshot();
});

test('Product with more information available renders correctly', () => {
  expect(
    shallow(<Product product={stubData.products.withMoreInformation} />)
  ).toMatchSnapshot();
});

test('Product without more information available renders correctly', () => {
  expect(
    shallow(<Product product={stubData.products.withoutMoreInformation} />)
  ).toMatchSnapshot();
});

test('Product with highlighted set renders correctly', () => {
  expect(
    shallow(<Product product={stubData.products.withHighlight} />)
  ).toMatchSnapshot();
});

test('Clicking the apply button displays the confirmation dialog', () => {
  const wrapper = shallow(
    <Product product={stubData.products.withConfirmation} />
  );

  wrapper.find(ApplyButton).simulate('click', { preventDefault() {} });
  expect(wrapper).toMatchSnapshot();
});

test('Clicking the more info button displays more information', () => {
  const wrapper = shallow(
    <Product product={stubData.products.withMoreInformation} />
  );

  wrapper.find(Button).simulate('click');
  expect(wrapper).toMatchSnapshot();
});
