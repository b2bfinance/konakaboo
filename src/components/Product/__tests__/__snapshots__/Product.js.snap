// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplyButton /> renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.15rem;
  box-sizing: border-box;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.75rem 1.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  class="c0"
>
  
</a>
`;

exports[`<Container /> renders correctly 1`] = `
.c0 {
  background-color: white;
  border: 2px solid #eceff1;
  margin-bottom: 1.5rem;
}

<div
  class="c0"
>
  
</div>
`;

exports[`<HeadingRow /> renders correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #eceff1;
  padding: 0.5rem 0.75rem;
}

<div
  class="c0 c1"
>
  
</div>
`;

exports[`<HighlightPoint /> renders correctly 1`] = `
.c0 {
  font-size: 1.5rem;
  font-weight: 400;
}

@media (min-width:62em) {
  .c0 {
    padding-right: 1rem;
  }
}

<li
  class="c0"
>
  
</li>
`;

exports[`<InfoList /> renders correctly 1`] = `
.c0 {
  list-style: none;
  margin-bottom: 0;
  margin-top: 1rem;
  padding: 0;
}

<ul
  class="c0"
>
  
</ul>
`;

exports[`<MoreInfoRow /> renders correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.c0 {
  border-top: 2px solid #eceff1;
  padding: 1rem;
  display: none;
}

<div
  class="c0 c1"
>
  
</div>
`;

exports[`<Product /> with highlighted set renders correctly 1`] = `
<styled.div
  data-product={true}
  highlight={true}
>
  <Styled(styled.div)
    highlight={true}
  >
    <pure(Star) />
    Landbay - Buy To Let Wizards
    <Component
      labels={
        Array [
          "broker only",
          "consumer favourite",
        ]
      }
    />
  </Styled(styled.div)>
  <styled.div>
    <Styled(styled.div)
      desktop="20"
      phone="100"
    >
      <img
        alt="Landbay"
        src="https://b2bfinance.com/images/partners/Landbay.png"
      />
    </Styled(styled.div)>
    <Component
      columns={
        Array [
          Object {
            "label": "Max LTV",
            "value": "85%",
          },
          Object {
            "label": "Initial Rate",
            "value": "1.49%",
          },
          Object {
            "label": "Subsequent Rate",
            "value": "2.75%",
          },
          Object {
            "label": "Fees",
            "value": "£1,495",
          },
        ]
      }
    />
    <Styled(styled.div)
      desktop="20"
      phone="100"
    >
      <Styled(styled.a)
        href="https://b2bfinance.com/enquire/buy-to-let"
        margin={true}
        onClick={[Function]}
        primary={true}
        rel="noopener"
        target="_blank"
      >
        Get Deal
      </Styled(styled.a)>
      <ProductConfirm
        description="B2Bfinance.com are registered intermediaries who will quote, package and submit your lending request to Natwest. There is no charge for this service as we are paid a small commission by Natwest to cover our time."
        forwardUrl="https://b2bfinance.com/enquire/buy-to-let"
        handleRequestClose={[Function]}
        open={false}
        title="Application via Brokers only"
      />
      <styled.a
        onClick={[Function]}
        secondary={true}
        slim={true}
      >
        more info
      </styled.a>
    </Styled(styled.div)>
  </styled.div>
  <Styled(styled.div)
    visible={false}
  >
    <span>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean bibendum facilisis condimentum. Donec laoreet porttitor condimentum. Nulla luctus efficitur magna, eget rhoncus metus placerat sed.
    </span>
    <styled.div>
      <styled.div
        desktop="50"
        phone="100"
      >
        <styled.ul>
          <Component
            points={
              Array [
                "Online Application",
                "Mobile app",
                "Customer service winner",
              ]
            }
          />
        </styled.ul>
      </styled.div>
      <styled.div
        desktop="50"
        phone="100"
      >
        <styled.ul>
          <Component
            points={
              Array [
                "1.5% Early Repayment",
                "6 months cool off peroid",
                "No monthly fee for existing customers",
              ]
            }
          />
        </styled.ul>
      </styled.div>
    </styled.div>
  </Styled(styled.div)>
</styled.div>
`;

exports[`<Product /> with more information available renders correctly 1`] = `
<styled.div
  data-product={true}
  highlight={false}
>
  <Styled(styled.div)
    highlight={false}
  >
    Precise Oh So Precise Total Winners Deal
    <Component
      labels={
        Array [
          "broker only",
        ]
      }
    />
  </Styled(styled.div)>
  <styled.div>
    <Styled(styled.div)
      desktop="20"
      phone="100"
    >
      <img
        alt="Barclays"
        src="https://b2bfinance.com/images/partners/Precise.png"
      />
    </Styled(styled.div)>
    <Component
      columns={
        Array [
          Object {
            "label": "Max LTV",
            "value": "80%",
          },
          Object {
            "label": "Initial Rate",
            "value": "1.75%",
          },
          Object {
            "label": "Subsequent Rate",
            "value": "3%",
          },
          Object {
            "label": "Fees",
            "value": "£995",
          },
        ]
      }
    />
    <Styled(styled.div)
      desktop="20"
      phone="100"
    >
      <Styled(styled.a)
        href="https://b2bfinance.com/enquire/buy-to-let"
        margin={true}
        onClick={[Function]}
        primary={true}
        rel="noopener"
        target="_blank"
      >
        Get Deal
      </Styled(styled.a)>
      <styled.a
        onClick={[Function]}
        secondary={true}
        slim={true}
      >
        more info
      </styled.a>
    </Styled(styled.div)>
  </styled.div>
  <Styled(styled.div)
    visible={false}
  >
    <span>
      Whether your landlord is a first time buyer, a first time landlord or has a large portfolio, we have a range of products to help. Our range of mortgages also includes products for Limited Companies and HMOs.
    </span>
    <styled.div>
      <styled.div
        desktop="50"
        phone="100"
      >
        <styled.ul>
          <Component
            points={
              Array [
                "Complete online application",
                "Mobile app",
                "Customer service winner",
              ]
            }
          />
        </styled.ul>
      </styled.div>
      <styled.div
        desktop="50"
        phone="100"
      >
        <styled.ul>
          <Component
            points={
              Array [
                "Rental cover - ICR: based on individual circumstances",
                "5 year fixed rate products that help maximise the loan size available",
                "5 year fixed with 4% cashback available on purchases",
              ]
            }
          />
        </styled.ul>
      </styled.div>
    </styled.div>
  </Styled(styled.div)>
</styled.div>
`;

exports[`<Product /> without more information available renders correctly 1`] = `
<styled.div
  data-product={true}
  highlight={false}
>
  <Styled(styled.div)
    highlight={false}
  >
    Natwest Low Low Rate
    <Component
      labels={Array []}
    />
  </Styled(styled.div)>
  <styled.div>
    <Styled(styled.div)
      desktop="20"
      phone="100"
    >
      <img
        alt="Natwest"
        src="https://b2bfinance.com/images/partners/NatWest.png"
      />
    </Styled(styled.div)>
    <Component
      columns={
        Array [
          Object {
            "label": "Max LTV",
            "value": "85%",
          },
          Object {
            "label": "Initial Rate",
            "value": "1.52%",
          },
          Object {
            "label": "Subsequent Rate",
            "value": "3.25%",
          },
          Object {
            "label": "Fees",
            "value": "£995",
          },
        ]
      }
    />
    <Styled(styled.div)
      desktop="20"
      phone="100"
    >
      <Styled(styled.a)
        href="https://b2bfinance.com/enquire/buy-to-let"
        margin={false}
        onClick={[Function]}
        primary={true}
        rel="noopener"
        target="_blank"
      >
        Get Deal
      </Styled(styled.a)>
      <ProductConfirm
        description="B2Bfinance.com are registered intermediaries who will quote, package and submit your lending request to Natwest. There is no charge for this service as we are paid a small commission by Natwest to cover our time."
        forwardUrl="https://b2bfinance.com/enquire/buy-to-let"
        handleRequestClose={[Function]}
        open={false}
        title="Application via Brokers only"
      />
    </Styled(styled.div)>
  </styled.div>
</styled.div>
`;

exports[`<ProductCol /> renders correctly 1`] = `
.c1 {
  width: 100%;
  min-height: 1px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 1.25rem;
  text-align: center;
}

.c0 img {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 100%;
  max-height: 120px;
}

.c0 span {
  font-size: 1.25rem;
  font-weight: 400;
  margin-bottom: 0.5rem;
  line-height: 1.1;
}

.c0 small {
  font-size: 95%;
  font-weight: 300;
}

<div
  class="c0 c1"
>
  
</div>
`;

exports[`<ProductColumns /> renders correctly 1`] = `
.c1 {
  width: 100%;
  min-height: 1px;
  -webkit-flex: 0 0 100%;
  -ms-flex: 0 0 100%;
  flex: 0 0 100%;
  max-width: 100%;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 1.25rem;
  text-align: center;
}

.c0 img {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 100%;
  max-height: 120px;
}

.c0 span {
  font-size: 1.25rem;
  margin-bottom: 0.5rem;
  line-height: 1.1;
}

.c0 small {
  font-size: 95%;
}

@media (min-width:62em) {
  .c1 {
    -webkit-flex: 0 0 60%;
    -ms-flex: 0 0 60%;
    flex: 0 0 60%;
    max-width: 60%;
  }
}

<div
  class="c0 c1"
>
  <small>
    test
  </small>
  <span>
    test
  </span>
</div>
`;

exports[`<ProductHighlightPoints /> renders correctly 1`] = `
.c0 {
  font-size: 1.5rem;
}

@media (min-width:62em) {
  .c0 {
    padding-right: 1rem;
  }
}

<li
  class="c0"
>
  test
</li>
`;

exports[`<ProductLabels /> renders correctly 1`] = `
.c1 {
  border: 1px solid;
  border-radius: 1rem;
  font-size: 80%;
  padding: 0.2rem 0.75rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 div {
  margin: .5rem .5rem 0 0;
}

@media (min-width:36em) {
  .c0 {
    margin-left: auto;
  }

  .c0 div {
    margin: 0 .5rem 0 0;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    test
  </div>
</div>
`;

exports[`<ProductTechnicalPoints /> renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.2rem;
  margin-bottom: 1rem;
}

.c0 svg {
  margin-right: 0.5rem;
}

@media (min-width:62em) {
  .c0 {
    margin-bottom: .5rem;
    padding-left: 1rem;
  }
}

<li
  class="c0"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root-1"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    >
      
    </path>
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
    >
      
    </path>
  </svg>
  test
</li>
`;

exports[`<StyledProductLabels /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 div {
  margin: .5rem .5rem 0 0;
}

@media (min-width:36em) {
  .c0 {
    margin-left: auto;
  }

  .c0 div {
    margin: 0 .5rem 0 0;
  }
}

<div
  class="c0"
>
  
</div>
`;

exports[`<TechnicalPoint /> renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.2rem;
  margin-bottom: 1rem;
}

.c0 svg {
  fill: #4dc16f;
  margin-right: 0.5rem;
}

@media (min-width:62em) {
  .c0 {
    margin-bottom: .5rem;
    padding-left: 1rem;
  }
}

<li
  class="c0"
>
  
</li>
`;
